File: transform/transform/others/tools/obfuscators/js-obfuscator/transform.js
const jsObfuscator = require('js-obfuscator');
const Utils = require("../../../utils")
const fs = require("fs");

/**
 * Directory for the configuration files
 */
const configDir = "./tools/obfuscators/js-obfuscator/configurations/" ;

 /**
 * Obfuscate a given code snippet with the js-obfuscator obfuscator. Stores the ouput in another file.
 * @param {*} input Code snippet to be transformed
 * @param {*} configFile Configuration files
 */
async function transform(input, configFile, fileId, suffix, outputDir){
    const config = Utils.load_config(configDir, configFile);
    await jsObfuscator(input, config).then(function(obfuscated) {
        //console.log(obfuscated)
        const fileDir = Utils.build_output_dir(fileId, outputDir, suffix, configFile);
        if(obfuscated == "undefined")
        {
            console.error("undefined")
        } else {
            Utils.store(fileDir, obfuscated); 
            console.log(fileDir)
        }
        }, function(error) {
            console.error(error)
        });
}
 


const args = process.argv.slice(2);
const file = args[0]
const configFile = args[1]
const suffix = args[2]
const outputDir = args[3]
const input = fs.readFileSync(file, "utf-8");
const fileId = Utils.get_fileId(file);
transform(input, configFile, fileId, suffix, outputDir)

File: transform/transform/others/tools/obfuscators/jsobfu/transform.rb
require 'jsobfu'
require 'date'
require "json"


CONFIG_DIR = "./tools/obfuscators/jsobfu/configurations/"

#Transform a snippet of code with the jsobfu tool, with all the specified configurations
# Params:
# +input+:: String containing the code to transform
# +fileId+:: Id of file transformed
# +configs+:: All the specified configurations for the jsobfu tool
def transform(inputFile, configFile)
  file = File.open(inputFile)
  input = file.read
  configFile = CONFIG_DIR + configFile
  config = File.read(configFile)
  hash = JSON.parse(config)
  configExtension = File.extname(configFile)
  configId = File.basename(configFile, configExtension) 
  iterations = hash['iterations']
  begin
    output = JSObfu.new(input).obfuscate(iterations:iterations)
    outputDir = "./output/obfuscated/jsobfu/"
    fileId = File.basename(inputFile, '.js')
    configId = File.basename(configFile, '.json')
    outputFile = outputDir + fileId + "_" + 'jsobfu' + "_" + configId + ".js"
    File.open(outputFile, 'w') { |file| file.write(output) }
  rescue IOError => e
    STDERR.puts e
  rescue SyntaxError => e
    STDERR.puts e
  rescue 
    STDERR.puts "Unknown reason."
  ensure
    file.close unless file.nil?
  end
end



transform(ARGV[0], ARGV[1])

